### PR TITLE
METRON-390: Stellar functions should initialize upon use, rather than all at once

### PR DESCRIPTION
**Pull request by cestella**
To test this, you should just spin up the environment in vagrant and everything should work as expected.

### CODE REVIEW

File: metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java [hunk 1]
```diff
@@ -188,15 +189,27 @@ public void initialize(Context context) {
       catch(Exception ex) {
         LOG.error(ex.getMessage(), ex);
       }
+      finally {
+        isInitialized = true;
```
Comment (nickwallen) [inline metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java]: If initialization fails, do we will want to set `isInitialized` to true?

File: metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java [hunk 2]
```diff
@@ -234,25 +248,36 @@ public Object apply(List<Object> args, Context context) throws ParseException {
...
+          return;
+        }
+        try {
+          MaaSConfig config = ConfigUtil.INSTANCE.read(client, "/metron/maas/config", new MaaSConfig(), MaaSConfig.class);
+          discoverer = new ServiceDiscoverer(client, config.getServiceRoot());
+          discoverer.start();
+          context.addCapability(Context.Capabilities.SERVICE_DISCOVERER, () -> discoverer);
+          isValidState = true;
+        } catch (Exception e) {
+          LOG.error(e.getMessage(), e);
```
Comment (nickwallen) [inline metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java]: If initialization fails, we're just logging the error.  Does it not make sense to fail fast or make some noise?

File: metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java (hunk 1 above)
Comment (cestella) [inline metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java]: I'd prefer for initialization to not continue to retry for every message, but rather the initialization routine should throw an exception.

File: metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java (hunk 2 above)
Comment (cestella) [inline metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java]: Yeah, agreed.  Throwing an exception here.

File: metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java (hunk 1 above)
Comment (nickwallen) [inline metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java]: That makes sense.  Maybe a little comment to that effect, will prevent me from asking you this same question in a few months.  Ha.

**Comment by nickwallen** [comment]
+1 looks good

**Pull request closed without merging**